extensions/amp-next-page/0.1/next-page-service.js

```diff
@@ -467,24 +483,29 @@ export class NextPageService {
    *     Position of the current recommendation unit in the viewport.
    */
   positionUpdate_(i, position) {
+    const isLastArticle = i === this.nextArticle_;
     // We're only interested when the recommendations exit the viewport
-    if (!position || position.positionRect !== null) {
+    if ((!position || position.positionRect !== null) && !isLastArticle) {
```

Comment:
Why do we exit early here it is not "isLastArticle"? It seems like the previous code would have triggered an analytics event during position updates when you say go from article 3 -> 2 -> 1 , but this will no longer do so when going from 2 -> 1 in that sequence.

Reply:
This is actually less restricted than the previous check because it allows the case where this is the last article but it does not have a `positionRect`. `positionRect === null` is used here as synonymous to `position.relativePos == 'top' || position.relativePos == 'bottom'` because `positionRect` is only specified if the `relativePos` is `'inside'`